zerver/lib/test_classes.py

```diff
@@ -2500,13 +2503,21 @@ def send_and_test_private_message(
 
         return msg
 
-    def build_webhook_url(self, *args: str, **kwargs: str) -> str:
-        url = self.URL_TEMPLATE
+    def build_webhook_url(self, *args: str, url_template: str | None = None, **kwargs: str) -> str:
+        if self.LEGACY_URL_TEMPLATES is not None and url_template is None:
+            for legacy_url_template in self.LEGACY_URL_TEMPLATES:
+                self.legacy_urls.append(
+                    self.build_webhook_url(*args, url_template=legacy_url_template, **kwargs)
+                )
+            index = len(self.legacy_urls) // 2
```

Comment:
In this recursion, for every legacy url template, we build the final url, and append at the end.
So, we end up with 2x elements in the list.
And we prune the first half, once we're done recursing.

I'm not sure this is readable and good code. Improvement suggestions / alternative implementations welcome.